puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/infra/updateJobs.groovy

```diff
@@ -0,0 +1,54 @@
+pipeline {
+    agent { label 'fast' }
+
+    stages {
+        stage('Setup workspace') {
+            steps {
+                checkout([
+                    $class : 'GitSCM',
+                    branches : [[name: 'master']],
+                    extensions: [[$class: 'CleanCheckout']],
+                    userRemoteConfigs: [
+                        [url: 'https://github.com/theforeman/foreman-infra.git']
+                    ]
+                ])
+
+                dir("./ci") {
+                    sh """
+                    virtualenv update_jobs
+                    source update_jobs/bin/activate
+                    pip install jenkins-job-builder
+                    deactivate
+                    """
+                }
+              }
+        }
+
+        stage('Update ci.centos.org jobs') {
+            steps {
+                dir("./ci") {
+                    withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'jenkins-centos', passwordVariable: 'PASSWORD', usernameVariable: 'USERNAME']]) {
+                        sh """
+                        source update_jobs/bin/activate
+                        mv ./centos.org/centos_jenkins.ini.example ./centos.org/jenkins_jobs.ini
+                        jenkins-jobs --conf ./centos.org/jenkins_jobs.ini --user ${env.USERNAME} --password ${env.PASSWORD} update --delete-old -r ./centos.org
+                        deactivate
+                        """
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        always {
+            script {
+                if(fileExists('ci/update_jobs')) {
```

Comment:
Just a thought: alternative to deleting the venv I wonder if it makes sense to install an explicit JJB version. We'd know what JJB we get and it would be a bit more efficient.

Reply:
Would this be along the lines of having a virtual env intstalled on the jenkins system itself?

Reply:
No, but a workspace could reuse virtualenvs. On the other hand, it could be a source of problems as well so I'd be fine with leaving that optimization out.